src/pynxtools_xps/reader.py

```diff
@@ -72,439 +72,575 @@
-        ):
-            if str(config_value).startswith(XPS_DATA_TOKEN):
-                key_part = config_value.split(XPS_DATA_TOKEN)[-1]
-                entries_values = find_entry_and_value(
-                    xps_data_dict, key_part, dt_typ=XPS_DATA_TOKEN
-                )
-                fill_data_group(
-                    key, key_part, entries_values, config_dict, xps_data_dict, template
-                )
+                        modified_key = config_key.replace(
+                            f"[{original_key}]",
+                            f"[{name}]",
+                        )
 
-            elif str(config_value).startswith(XPS_DETECTOR_TOKEN):
-                key_part = config_value.split(XPS_DETECTOR_TOKEN)[-1]
-                entries_values = find_entry_and_value(
-                    xps_data_dict, key_part, dt_typ=XPS_DETECTOR_TOKEN
-                )
+                        if (
+                            isinstance(config_value, str)
+                            and f"{original_key}/" in config_value
+                        ):
+                            modified_value = config_value.replace(
+                                f"{original_key}/", f"{name}/"
+                            )
 
-                fill_detector_group(
-                    key, entries_values, config_dict, xps_data_dict, template
-                )
+                        self.config_dict[modified_key] = modified_value
+                        del self.config_dict[config_key]
 
-            elif str(config_value).startswith(XPS_TOKEN):
-                key_part = config_value.split(XPS_TOKEN)[-1]
-                entries_values = find_entry_and_value(
-                    xps_data_dict, key_part, dt_typ=XPS_TOKEN
-                )
-                for entry, ent_value in entries_values.items():
-                    modified_key = key.replace("/ENTRY[entry]/", f"/ENTRY[{entry}]/")
-                    fill_template_with_value(modified_key, ent_value, template)
+    def get_metadata(
+        self,
+        metadata_dict: Dict[str, Any],
+        path: str,
+        entry_name: str,
+    ) -> Any:
+        """
+        Get metadata from the ELN or XPS data dictionaries.
+
+        Note that the keys of metadata_dict may contain more than
+        the path, i.e.,
+        /ENTRY[my-entry]/instrument/analyser/collectioncolumn/voltage.
+        With the regex, the path = "collectioncolumn/voltage" would
+        still yield the correct value.
+
+        Parameters
+        ----------
+        metadata_dict : Dict[str, Any]
+            One of ELN or XPS data dictionaries .
+        path : str
+            Path to search in the metadata_dict
+        entry_name : str
+            Entry name to search.
+
+        Yields
+        ------
+        value: Any
+            The value in the metadata_dict.
+
+        """
+        pattern = re.compile(
+            rf"^/ENTRY\[{re.escape(entry_name)}\](?:/.*/|/){re.escape(path)}$"
```

Comment:
Nice didn't know that there is an escape function. (This is no real comment, just praise :)

Reply:
Thanks :)